core/trino-main/src/main/java/io/trino/server/remotetask/HttpRemoteTask.java

```diff
@@ -136,6 +136,10 @@
     private final SetMultimap<PlanNodeId, ScheduledSplit> pendingSplits = HashMultimap.create();
     private final int maxUnacknowledgedSplits;
     @GuardedBy("this")
+    private volatile int queueSizeAdjustment;
```

Comment:
this never goes down, right?

Reply:
The value of this parameter is used separately for every stage, so after scheduling is completed, reducing it has no impact on how tasks are processed in other stage. But if you think this need to goes down, I will try to look for some example where this could be beneficial.